### DISCUSSION TITLE
some errors when I run hover_swarm.launch

### DISCUSSION THREAD

**Discussion by lwisle** [USC-ACTLab/crazyswarm#490] [Q&A]
Hello

I want to run hover_swarm.launch with crazyflie and optitrack. The Ubuntu version is 20.04, with python 3.8.10. The motive's version is 1.9. My motive IP is: 192.168.2.228, and my Ubuntu connect optitrack by the same wifi. I have refresh crazyflie's firmware and radio PA's. But when I run the hover_swarm.launch and mocap_helper.launch,the errors occur as follow:

Firstly, I run hover_swarm.launch with motive setting as follow:
![Screenshot (2)](https://user-images.githubusercontent.com/82260604/135738374-261bf34f-d765-4751-845d-878f5f0c32e5.png)
and the result is here:I don't know why there is no image in Rviz and the crazyflie didn't take off.
![2021-10-03 11-09-53屏幕截图](https://user-images.githubusercontent.com/82260604/135738386-c6cc147f-d784-46ed-b125-7338a8a7e7b9.png)
the crazyswarm settings are here:
![2021-10-03 11-10-37屏幕截图](https://user-images.githubusercontent.com/82260604/135738441-ab062899-59a8-4f1c-8819-9da71235e4b5.png)

Secondly, I try mocap_helper.launch,the result is here:
![2021-10-03 11-12-06屏幕截图](https://user-images.githubusercontent.com/82260604/135738463-363b8d27-1af9-403c-aa08-45cdd8d8c936.png)

Then I reset the motive settings and try again hover_swarm.launch:
![Screenshot (3)](https://user-images.githubusercontent.com/82260604/135738487-d35cb8ce-4ff0-4efb-9a5d-401a72460dfa.png)
![2021-10-03 11-13-05屏幕截图](https://user-images.githubusercontent.com/82260604/135738476-fe9ec406-b026-4243-ac42-23d2bbe85cbf.png)

Finally, I try NatNetSDK:
![2021-10-03 11-14-53屏幕截图](https://user-images.githubusercontent.com/82260604/135738513-f7dcaa05-ecfc-407e-ad87-3f4aa62b7598.png)
![2021-10-03 11-15-31屏幕截图](https://user-images.githubusercontent.com/82260604/135738519-44de8ebe-3a13-4b9a-aa44-df76399f39e1.png)
The results show command channel is well but test date couldn't run.

what can I do to fly crazyflie with crazyswarm ? I would appreciate your guidance,thanks.

**Comment by whoenig** [comment]
Would it be possible for you to upgrade to Motive 2.3 or 3.0? These are the versions we test with. From your output it looks likely to be a bug in parsing the datastream, but it will be difficult for us to fix if we can't reproduce the issue on our end.

**Reply by Frostholm12** [reply]
Helle @whoenig 

I am currently running into the same exact issue, with the same setup except I have motive 2.3.1.

do you have a suggestion as to how to solve this?

**Reply by whoenig** [reply]
If you can switch to Crazyswarm2, there is an option to use the official OptiTrack SDK which typically resolves these packet parsing issues.
